controllers/healthcheck_controller.go

```diff
@@ -471,7 +471,9 @@ func (r *HealthCheckReconciler) createSubmitWorkflow(ctx context.Context, log lo
 	workflow.SetGroupVersionKind(wfGvk)
 	workflow.SetNamespace(hc.Spec.Workflow.Resource.Namespace)
 	workflow.SetGenerateName(hc.Spec.Workflow.GenerateName)
+	r.TimerLock.RLock()
 	workflow.SetLabels(r.workflowLabels)
+	r.TimerLock.RUnlock()
```

Comment:
Shouldn't .RUnlock be deferred?
```suggestion
	defer r.TimerLock.RUnlock()
```

Reply:
not required in this case.. as the lock is only for workflowLabels and there is other piece of code after this that can continue to run in that method.